src/widgets/modals/FinaliseModal.js

```diff
@@ -63,16 +64,18 @@ export const FinaliseModal = props => {
   };
 
   return (
-    <ConfirmModal
-      isOpen={isOpen}
-      questionText={errorText || modalStrings[finaliseText]}
-      confirmText={modalStrings.confirm}
-      cancelText={errorText ? modalStrings.got_it : modalStrings.cancel}
-      onConfirm={!errorText ? tryFinalise : null}
-      onCancel={() => {
-        if (onClose) onClose();
-      }}
-    />
+    <ModalContainer fullScreen={true} isVisible={isOpen}>
```

Comment:
I noticed that now the `Spinner` is not being shown when finalising! I think we need to close the finalise modal before loading the spinner (through `runWithLoadingIndicator`) ? I think moving the `onClose` in the `tryFinalise` method to be before `runWithLoadingIndicator` should do the trick!

Reply:
@joshxg 
done. `c27d62c73`